src/main.leo

```diff
@@ -197,26 +211,28 @@ program privx_exchange_beta4.aleo {
 
         // Increase the balance of `to`
         let key_pair: Pair = Pair {
-            addr: to,
+            account: to_account,
             token_id: token_id
         };
         let current_amount: u64 = Mapping::get_or_use(balance, key_pair, 0u64);
         Mapping::set(balance, key_pair, current_amount + amount);
     }
 
     transition transfer_public(public token_id: u64, public to: address, public amount: u64) {
-        return then finalize(token_id, self.caller, to, amount);
+        let from_account: field = default_hash_account(self.caller);
```

Comment:
Besides transition tranferring from public to private there should be other ones like:
transition transfer_public_to_private(owner: address, value: u128) -> Token
transition transfer_private_to_public(t: Token) -> bool
transition combine_token_records(t1: Token, t2: Token) -> t: Token
transition split_token_records(t: Token) -> t1: Token, t2: Token